Tests fail sporadically with error "IO Failed to bind to 0.0.0.0/0.0.0.0:XXXX"

### Describe the bug

Occasionally, an error occurs during the build process of Opencast when running tests. The error message can be found here: [GitHub Actions Log](https://github.com/opencast/opencast/actions/runs/9597736269/job/26467429541#step:7:7887).

### To Reproduce

Steps to reproduce the behavior:
1. Initiate the build process for Opencast. (Create a PR for example)
2. Observe the error (note: it does not always appear).

### Expected behavior

The build process and tests should complete successfully without errors.

### Stack trace
See the error message linked above for details.

### Additional context

The error is hard to reproduce as it seems to be caused by an already occupied server port, which is randomly assigned.

Since the GitHub Action logs are not preserved forever, here is the relevant error from the logs:

```

Error:  Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.052 s <<< FAILURE! -- in org.opencastproject.adminui.endpoint.AclEndpointTest
Error:  org.opencastproject.adminui.endpoint.AclEndpointTest -- Time elapsed: 0.049 s <<< ERROR!
java.io.IOException: Failed to bind to 0.0.0.0/0.0.0.0:58344
	at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:349)
	at org.eclipse.jetty.server.ServerConnector.open(ServerConnector.java:310)
	at org.eclipse.jetty.server.AbstractNetworkConnector.doStart(AbstractNetworkConnector.java:80)
	at org.eclipse.jetty.server.ServerConnector.doStart(ServerConnector.java:234)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
	at org.eclipse.jetty.server.Server.doStart(Server.java:401)
	at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:73)
	at org.opencastproject.test.rest.RestServiceTestEnv.setUpServer(RestServiceTestEnv.java:130)
	at org.opencastproject.adminui.endpoint.AclEndpointTest.oneTimeSetUp(AclEndpointTest.java:58)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunBefores.invokeMethod(RunBefores.java:33)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:316)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:240)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:214)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:155)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)
Caused by: java.net.BindException: Address already in use
	at java.base/sun.nio.ch.Net.bind0(Native Method)
	at java.base/sun.nio.ch.Net.bind(Net.java:459)
	at java.base/sun.nio.ch.Net.bind(Net.java:448)
	at java.base/sun.nio.ch.ServerSocketChannelImpl.bind(ServerSocketChannelImpl.java:227)
	at java.base/sun.nio.ch.ServerSocketAdaptor.bind(ServerSocketAdaptor.java:80)
	at org.eclipse.jetty.server.ServerConnector.openAcceptChannel(ServerConnector.java:344)
	... 28 more

[…]

[INFO] Results:
[INFO] 
Error:  Errors: 
Error:    AclEndpointTest.oneTimeSetUp:58 » IO Failed to bind to 0.0.0.0/0.0.0.0:58344
```

--- a/modules/rest-test-environment/src/main/java/org/opencastproject/test/rest/RestServiceTestEnv.java
+++ b/modules/rest-test-environment/src/main/java/org/opencastproject/test/rest/RestServiceTestEnv.java
@@ -33,7 +33,7 @@ import org.glassfish.jersey.servlet.ServletContainer;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.net.BindException;
+import java.io.IOException;
 import java.net.URL;
 import java.util.concurrent.ThreadLocalRandom;
 
@@ -117,28 +117,33 @@ public final class RestServiceTestEnv {
    * Call in @BeforeClass annotated method.
    */
   public void setUpServer() {
+    int port = -1;
+    int maxRetries = 100;
     try {
-      for (int tries = 100; tries > 0; tries--) {
+      ServletContainer servletContainer = new ServletContainer(cfg);
+      ServletHolder jerseyServlet = new ServletHolder(servletContainer);
+      for (int tries = maxRetries; tries > 0; tries--) {
         try {
-          final int port = 3000 + tries + ThreadLocalRandom.current().nextInt(62000);
-          logger.error("Start http server at port {}", port);
+          port = 3000 + tries + ThreadLocalRandom.current().nextInt(62000);
+          logger.info("Starting http server at port {}", port);
           server = new Server(port);
-          ServletContainer servletContainer = new ServletContainer(cfg);
-          ServletHolder jerseyServlet = new ServletHolder(servletContainer);
           ServletContextHandler context = new ServletContextHandler(server, "/");
           context.addServlet(jerseyServlet, "/*");
           server.start();
           baseUrl = UrlSupport.url("http", "127.0.0.1", port);
           return;
-        } catch (BindException e) {
+        } catch (IOException e) {
+          logger.error("Couldn't bind server to port {}. Retrying with new port...", port, e);
           // Rethrow exception after last try
           if (tries == 1) {
+            logger.error("Couldn't start server after {} tries.", maxRetries);
             throw e;
           }
           Thread.sleep(100);
         }
       }
     } catch (Exception e) {
+      logger.error("Unexpected Exception occurred while setting up http server");
       chuck(e);
     }
   }

